Prox/Prox/Data/PlacesController.swift

```diff
@@ -168,14 +201,17 @@ class PlacesProvider {
     private func displayPlaces(places: [Place], forLocation location: CLLocation) {
         let filteredPlaces = PlaceUtilities.filterPlacesForCarousel(places)
         return PlaceUtilities.sort(places: filteredPlaces, byTravelTimeFromLocation: location, ascending: true, completion: { sortedPlaces in
+            self.placesLock.lock()
             self.places = sortedPlaces
             DispatchQueue.main.async {
                 self.delegate?.placesProvider(self, didReceivePlaces: self.places)
+                self.placesLock.unlock()
```

Comment:
From the docs:

> The NSLock class uses POSIX threads to implement its locking behavior. When sending an unlock message to an NSLock object, you must be sure that message is sent from the same thread that sent the initial lock message. Unlocking a lock from a different thread can result in undefined behavior.

Here, we're likely unlocking from a different thread.

Hack fix: Since we never directly modify the places array (but rather reassign the reference), lock, make a copy of the reference, unlock, and pass the copy to `didReceivePlaces` instead.